flake8_isort.py

```diff
@@ -1,16 +1,67 @@
 # -*- coding: utf-8 -*-
+from __future__ import print_function
+from contextlib import contextmanager
 from isort import SortImports
 
+import sys
+
+
+class DummyFile(object):
+    def write(self, x): pass
+
+
+@contextmanager
+def supress_stdout():
+    save_stdout = sys.stdout
+    sys.stdout = DummyFile()
+    yield
+    sys.stdout = save_stdout
+
+
+class Read(object):
+
+    @classmethod
+    def file(cls, filename):
+        if filename == 'stdin':
+            return None
+        with open(filename, 'r') as f:
+            return f.read()
+
+    @classmethod
+    def tree(cls, tree):
+        return getattr(tree, 'source', None)
+
+    @classmethod
+    def stdin(cls):
+        content = sys.stdin.read()
+        if not content:
+            # issue: https://github.com/PyCQA/pep8/pull/443
+            print('Flake8Isort: stdin is empty')
+        return content
+
 
 class Flake8Isort(object):
     name = 'flake8_isort'
     version = '0.1'
     message = 'I001 found unsorted imports'
 
-    def __init__(self, tree, filename):
-        self.filename = filename
+    def __init__(self, tree, filename, builtins=None):
+        self.content = Read.tree(tree) or Read.file(filename) or Read.stdin()
```

Comment:
I would rather prefer to have something like:
``self.content = Read(tree, filename)``

and returns the expected value, i.e. hiding the complexity inside ``Read`` instead of having it on ``Flake8Isort``.